src/LibraryViewExtension/LibraryViewCustomization.cs

```diff
@@ -0,0 +1,126 @@
+        /// operation will fail if the current specification already contains a 
+        /// section with the same text as of the given section. The clients must
+        /// call AddElements method to add layout elements to a given section.
+        /// </summary>
+        public bool AddSections(IEnumerable<LayoutSection> sections)
+        {
+            var spec = GetSpecification();
+            spec.sections.AddRange(sections);
+            var count = spec.sections.Count;
+            if (spec.sections.GroupBy(s => s.text).Count() < count) return false; //some duplicate sections exist
+
+            //Update the specification
+            SetSpecification(spec);
+            return true;
+        }
+
+        /// <summary>
+        /// Allows clients to add a list of elements to a given section. This 
+        /// operation will fail if the text property of any of the child elements 
+        /// of the given section matches with the text property of the given 
+        /// elements.
+        /// </summary>
+        public bool AddElements(IEnumerable<LayoutElement> elements, string sectionText = "")
+        {
+            var text = DefaultSectionName;
+            if (!string.IsNullOrEmpty(sectionText))
+            {
+                text = sectionText;
+            }
+            var spec = GetSpecification();
+            var section = spec.sections.FirstOrDefault(x => string.Equals(x.text, text));
+            if (section == null)
+            {
+                section = new LayoutSection(text);
+                spec.sections.Add(section);
+            }
+            section.childElements.AddRange(elements);
+            int count = section.childElements.Count;
+            if (section.childElements.GroupBy(e => e.text).Count() < count) return false;
+
+            //Update the specification
+            SetSpecification(spec);
+            return true;
+        }
+
+        /// <summary>
+        /// Allows clients to add a list of include path to a given section. This 
+        /// operation will fail if the path property of any of the includes 
+        /// of the given section conflicts with the path property of the given 
+        /// includes.
+        /// </summary>
+        public bool AddIncludeInfo(IEnumerable<LayoutIncludeInfo> includes, string sectionText = "")
+        {
+            var text = DefaultSectionName;
+            if (!string.IsNullOrEmpty(sectionText))
+            {
+                text = sectionText;
+            }
+            var spec = GetSpecification();
+            var section = spec.sections.FirstOrDefault(x => string.Equals(x.text, text));
+            if (section == null)
+            {
+                section = new LayoutSection(text);
+                spec.sections.Add(section);
+            }
+            section.include.AddRange(includes);
+            int count = section.include.Count;
+            if (section.include.GroupBy(e => e.path).Count() < count) return false;
```

Comment:
What does it mean when this happens? User error? If this is taken as a user error, then better error message should be there like which includes are duplicated.